frontend/src/layout/RTCLayout.tsx

```diff
@@ -0,0 +1,60 @@
+import { useCallback, useEffect } from "react";
+import { Outlet } from "react-router-dom";
+import { useRecoilValueLoadable, useSetRecoilState } from "recoil";
+import { ensembleAtom, userListAtom } from "../recoil/ensemble";
+import { socketAtom } from "../recoil/socket";
+import { Ensemble, EnsembleObject } from "common/dist";
+
+/**
+ * This is a separate layout for Real Time Collaboration (RTC)
+ *
+ * When navigating to the Ensemble view, this component gets loaded.
+ * When this happens, we communicate with the server to set up the necessary parts
```

Comment:
I'd change "necessary parts" to "necessary listeners" so that the listeners are ready to catch anything the server sends our way.